src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostOnAutoInsertEndpointTest.cs

```diff
@@ -0,0 +1,275 @@
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostOnAutoInsertEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Theory]
+    [InlineData("PageTitle")]
+    [InlineData("div")]
+    [InlineData("text")]
+    public async Task Component_AutoInsertEndTag(string startTag)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}>$$
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <{startTag}>$0</{startTag}>
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: ">");
+    }
+
+    [Theory]
+    [InlineData("PageTitle")]
+    [InlineData("div")]
+    [InlineData("text")]
+    public async Task Component_DoNotAutoInsertEndTag_DisabledAutoClosingTags(string startTag)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}>$$
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <{startTag}>
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: ">", autoClosingTags: false, expectResult: false);
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertAttributeQuotes()
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <PageTitle style=$$></PageTitle>
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <PageTitle style="$0"></PageTitle>
+
+            The end.
+            """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "=", delegatedResponseText: "\"$0\"");
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertCSharp_OnForwardSlash()
+    {
+        var input = """
+        @code {
+            ///$$
+            void TestMethod() {}
+        }
+        """;
+
+        var output = """
+        @code {
+            /// <summary>
+            /// $0
+            /// </summary>
+            void TestMethod() {}
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "/");
+    }
+
+    [Fact]
+    public async Task Component_DoNotAutoInsertCSharp_OnForwardSlashWithFormatOnTypeDisabled()
+    {
+        var input = """
+        @code {
+            ///$$
+            void TestMethod() {}
+        }
+        """;
+
+        var output = """
+        @code {
+            ///
+            void TestMethod() {}
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "/", formatOnType: false, expectResult: false);
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertCSharp_OnEnter()
```

Comment:
I suspect this is just copy and paste, but almost none of these tests have anything to do with "Component", and the class name already tells us it's "AutoInsert", so these could all be trimmed.